client/my-sites/patterns/components/header/index.tsx

```diff
@@ -1,23 +1,180 @@
 import { preventWidows } from 'calypso/lib/formatting';
 import { PatternsSearchField } from 'calypso/my-sites/patterns/components/search-field';
+import type { Category } from 'calypso/my-sites/patterns/types';
 
 import './style.scss';
 
-type PatternsHeaderProps = {
-	description: string;
-	title: string;
+const OPTIONS = {
+	home: {
+		title: {
+			comment: 'HTML title of the page that showcases block patterns',
+			textOnly: true,
+		},
+		header: {
+			comment: 'Header of the page that showcases block patterns',
+			textOnly: true,
+		},
+		description: {
+			comment: 'Description of the page that showcases block patterns',
+			textOnly: true,
+		},
+	},
+	category: {
+		title: {
+			comment: 'HTML title of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+		header: {
+			comment: 'Header of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+		description: {
+			comment: 'Description of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+	},
 };
 
-export const PatternsHeader = ( { description, title }: PatternsHeaderProps ) => {
+const CONTENT: Record<
+	Category[ 'name' ],
+	{ description: Substitution; header: Substitution; title: Substitution }
+> = {
+	default: {
+		title: translate( 'WordPress Patterns', OPTIONS.home.title ),
+		header: translate( "It's easier with patterns", OPTIONS.home.header ),
+		description: translate(
+			'Dive into hundreds of expertly designed, fully responsive layouts, and bring any kind of site to life, faster.',
+			OPTIONS.home.description
+		),
+	},
+	header: {
+		title: translate( 'WordPress Header Patterns', OPTIONS.category.title ),
+		header: translate( 'Get a head start with patterns', OPTIONS.category.header ),
+		description: translate(
+			"Use predefined and beautiful patterns to decorate the header of your site, and catch your visitors' attention.",
+			OPTIONS.category.description
+		),
+	},
+	footer: {
+		title: translate( 'WordPress Footer Patterns', OPTIONS.category.title ),
+		header: translate( 'Build a strong foundation with patterns', OPTIONS.category.header ),
+		description: translate(
+			'Choose from our collection of patterns to add essential information and navigation to the bottom of your site.',
+			OPTIONS.category.description
+		),
+	},
+	about: {
+		title: translate( 'WordPress About Patterns', OPTIONS.category.title ),
+		header: translate( 'Tell your story with patterns', OPTIONS.category.header ),
+		description: translate(
+			'Add one of these patterns to your site to convey your mission and values compellingly.',
+			OPTIONS.category.description
+		),
+	},
+	posts: {
+		title: translate( 'WordPress Blog Post Patterns', OPTIONS.category.title ),
+		header: translate( 'Get more readers with patterns', OPTIONS.category.header ),
+		description: translate(
+			'Take advantage of this selection of patterns that make your writing pop and keep readers engaged.',
```

Comment:
```suggestion
			'Take advantage of this selection of patterns to display your latest blog posts, keeping readers engaged.',
```
I feel this description could be more to-the-point.